src/classes/RD2_ScheduleService.cls

```diff
@@ -37,6 +37,9 @@
 
 public without sharing class RD2_ScheduleService {
 
+    @TestVisible
+    private Date testCurrentDate = Date.today();
```

Comment:
I very much like this, though I recommend an implementation such as 
```
@TestVisible
private Date currentDate {
    get {
        if (currentDate == null) {
            currentDate = Date.today();
        }
        return currentDate;
    } set;
}
```

With the above, you won't need the ternary `if` on line 107.